ddl/ddl_api.go

```diff
@@ -843,6 +877,20 @@ func (d *ddl) AddColumn(ctx context.Context, ti ast.Ident, spec *ast.AlterTableS
 		return infoschema.ErrColumnExists.GenByArgs(colName)
 	}
 
+	// If new column is a generated column, do validation.
+	for _, option := range spec.NewColumn.Options {
+		if option.Tp == ast.ColumnOptionGenerated {
+			normalColNames := make(map[string]struct{}, len(t.Cols()))
+			for _, col := range t.Cols() {
+				normalColNames[col.Name.L] = struct{}{}
```

Comment:
**add the generated column at first position** now we don't support add columns at specified positions. we can only append them.

the name is bad.